src/components/ListOfGoods.jsx

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+
+class ListOfGoods extends React.Component{
+  state = {
+    isVisible: false,
+    isReversed: false,
+    sortBy: '',
+  }
+
+  showGoodsList = () => {
+    this.setState({
+      isVisible: true
+    })
+  }
+
+  reverseGoodsList = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed
+    }))
+  }
+
+  sortAlphabetically = () => {
+    this.setState({
+      sortBy: 'alphabet',
+    })
+  }
+
+  sortByLength = () => {
+    this.setState({
+      sortBy: 'length',
+    })
+  }
+
+  reset = () => {
+    this.setState({
+      isReversed: false,
+      sortBy: '',
+    })
+  }
+
+  render () {
+    const {goods} = this.props;
+    const {isVisible, isReversed, sortBy} = this.state;
+    const goodsCopy = [...goods];
+
+    goodsCopy.sort((a, b) => {
+      switch (sortBy) {
+        case 'alphabet':
+          return a.localeCompare(b);
+        case 'length':
+          return a.length - b.length;
+      }
+    })
+
+    if(isReversed) {
+      goodsCopy.reverse()
+    }
+
+    return (
+      <div>
+        {isVisible === false && (
+          <button
+            type="button"
+            onClick={this.showGoodsList}
+          >
+            Start
```

Comment:
Let's move this button and logic of rendering the list-component on the upper level - to the App. You should conditionally render your `GoodsList` component when clicking on this button at the `App`